scripts/combine/setupCombine.py

```diff
@@ -10,31 +10,35 @@
 def make_parser(parser=None):
     parser = argparse.ArgumentParser()
     parser.add_argument("-o", "--outfolder", type=str, default=".", help="Output folder with the root file storing all histograms and datacards for single charge (subfolder WMass or ZMassWLike is created automatically inside)")
-    parser.add_argument("-i", "--inputFile", type=str)
+    parser.add_argument("-i", "--inputFile", nargs="+", type=str)
     parser.add_argument("-p", "--postfix", type=str, help="Postfix for output file name", default=None)
     parser.add_argument("-v", "--verbose", type=int, default=3, choices=[0,1,2,3,4],
                         help="Set verbosity level with logging, the larger the more verbose")
     parser.add_argument("--noColorLogger", action="store_true", help="Do not use logging with colors")
+    parser.add_argument("--hdf5", action="store_true", help="Write out datacard in hdf5")
+    parser.add_argument("--sparse", action="store_true", help="Write out datacard in sparse mode (only for when using hdf5)")
     parser.add_argument("--excludeProcGroups", type=str, nargs="*", help="Don't run over processes belonging to these groups (only accepts exact group names)", default=["QCD"])
     parser.add_argument("--filterProcGroups", type=str, nargs="*", help="Only run over processes belonging to these groups", default=[])
     parser.add_argument("-x", "--excludeNuisances", type=str, default="", help="Regular expression to exclude some systematics from the datacard")
     parser.add_argument("-k", "--keepNuisances", type=str, default="", help="Regular expression to keep some systematics, overriding --excludeNuisances. Can be used to keep only some systs while excluding all the others with '.*'")
     parser.add_argument("--absolutePathInCard", action="store_true", help="In the datacard, set Absolute path for the root file where shapes are stored")
     parser.add_argument("-n", "--baseName", type=str, help="Histogram name in the file (e.g., 'nominal')", default="nominal")
     parser.add_argument("--noHist", action='store_true', help="Skip the making of 2D histograms (root file is left untouched if existing)")
-    parser.add_argument("--qcdProcessName" , type=str, default="Fake", help="Name for QCD process")
+    parser.add_argument("--qcdProcessName" , type=str, default=None, help="Name for QCD process")
```

Comment:
Attention here, this should stay explicitly as Fake (or whatever default name we have in datagroups2016.py, now it is Fake)

Reply:
Hmm, I definitely didn't do this, it must have been in something I merged in. Was this intentional @davidwalter2 ?

Reply:
Now I remember, it was intentional. The reason is that when fitting in the low PU electrons and muons together in a simultaneous ABCD fit we need two fake processes. Having a single string with "Fake" sets both to the same name. So I made the fakename a property of the datagroup and by None I wanted that it picks the right default name. 
I see that the fake name is not handled consistently, at some places we use still the string "Fake" on other places we get/set the fakename via the card tool, and some options like "--noStatUncFakes" don't work with having "--qcdProcessName" with None as default, ideally this could be cleaned up.

Reply:
Option --noStatUncFakes is probably no longer needed. In the past we wanted to test the case where we set the uncertainty to 0, but it is not a good configuration, since fluctuations in the template still exist.

About the rest, my impression is that different analyses have different needs, but once the analysis is chosen the name of samples must be unambiguously defined, so we could arrange it in such a way that when the default is None it is actually set to a specific analysis dependent value (while when it is set by the user it should be implied that it is intended to have that particular name and thus must override any internal default setting)